tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +174,640 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+
+/**
+ * Get Array Schema from file
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * tiledb_array_schema_t* schema;
+ * tiledb_file_get_schema(ctx, file, &schema);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param array_schema
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_get_schema(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    tiledb_array_schema_t** array_schema);
+
+/**
+ * Export a file to a raw buffer
+ *
+ * * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * const char* file_uri = "some_file";
+ * FILE* file_out = fopen(file_uri, "w+");
+ * tiledb_file_export_fh(ctx, file, file_out, NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param out FILE handle to write to
+ * @param config TileDB Config object for export settings
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_export_fh(
+    tiledb_ctx_t* ctx, tiledb_file_t* file, FILE* out, tiledb_config_t* config);
+
+/**
+ * Export a file to a raw buffer
+ *
+ * * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * uint64_t size = 0;
+ * uint64_t file_offset = 0;
+ * tiledb_file_get_size(ctx, file, &size);
+ * void* buffer = malloc(size);
+ * tiledb_file_export_buffer(ctx, file, bytes, &size, offset, NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param bytes output buffer, alloc'ed by used.
+ * @param size size to read
+ * @param file_offset file_offset to read from
+ * @param config TileDB Config object for export settings
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_export_buffer(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    void* bytes,
+    uint64_t* size,
+    uint64_t file_offset,
+    tiledb_config_t* config);
+
+/**
+ * Export a file to the provided URI
+ *
+ * * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * tiledb_file_export_uri(ctx, file, "s3://tiledb_bucket/some_output_file",
+ * NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param output_uri output uri to save to
+ * @param config TileDB Config object for export settings
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_export_uri(
```

Comment:
This is the only export function we need, so let's call it `tiledb_file_export`.